near/omni-bridge/src/lib.rs

```diff
@@ -875,6 +1022,14 @@ impl Contract {
     pub fn get_current_destination_nonce(&self, chain_kind: ChainKind) -> Nonce {
         self.destination_nonces.get(&chain_kind).unwrap_or_default()
     }
+
+    #[private]
+    pub fn ft_resolve_transfer(&mut self, amount: U128) -> U128 {
+        match env::promise_result(0) {
+            PromiseResult::Successful(_) => U128(0),
+            PromiseResult::Failed => amount,
```

Comment:
This refund is dangerous due to the undetermined behavior of the `ft_trnasfer_call`, so let's always return zero.

Reply:
Okay, just to confirm, if `ft_transfer_call` reverts it's sender who will lose money and not the relayer, right? So senders will be responsible for the target of `ft_transfer_call`

Reply:
Yes.
In the follow-up PR, we can try to solve the refund issue, which was mentioned multiple times in the audits.
What we should take into account is:
- The `ft_transfer_call` can fail only if the storage wasn't deposited or due to malicious contract, otherwise it returns.
- The mint call `ft_transfer_call` if there is a message, so it has the same behaviour.
- The `ft_transfer` can fail only if the storage wasn't deposited.

Since we already verify the storage deposit, then we can do the refund only if the `ft_transfer_call` returned a refund value.

Reply:
Just realized that no refund is more dangerous than I thought because it is possible to attach less gas to provoke the fail.